### PR TITLE
Letters 1 and 2 html templates merge

### PR DESCRIPTION
**Pull request by Miles-Alford**
Both the letters 1 and 2 html templates have been created

### CODE REVIEW

File: lib/hackney/pdf/templates/letter_1_in_arrears_LH_agreed_template.erb [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+  <p>
+    020 8326 2299 <br>
+    service.charges@hackney.gov.uk <br>
+    Your ref: <%= params.fetch(:payment_ref) %> <br>
+    Our ref:
+  </p>
+
+  <p>
+    Date: <%= Time.now.strftime("%d %B %Y") %>
```
Comment (mrosel) [inline lib/hackney/pdf/templates/letter_1_in_arrears_LH_agreed_template.erb]: Recommend not using `strftime` as this fragments config/code. Instead, use a configuration/translation object with i18n - see http://www.stackednotion.com/blog/2015/01/03/formatting-dates-and-times-in-rails-with-i18n-localize/ for more info.

File: app/models/hackney/service_charge/letter.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                    .map { |mandatory_field| { name: mandatory_field.to_s, message: 'missing mandatory field' } }
+
+        letter_params[:lessee_short_name] = letter_params[:lessee_full_name] unless letter_params[:lessee_short_name].present?
+        letter_params
+      end
+
+    end
+  end
+end
+
```
Comment (elena-vi) [inline app/models/hackney/service_charge/letter.rb]: @mrosel Wanna give me some feedback on this?

File: lib/hackney/pdf/preview.rb [hunk 3]
```diff
@@ -14,6 +14,8 @@ def execute(payment_ref:, template_id:)
           template_path: template[:path]
         ).execute(letter_params: leasehold_info)
 
+        Hackney::PDF::Generator.new.execute(preview_with_errors[:html]).to_file("#{Time.now}.pdf")
```
Comment (elena-vi) [inline lib/hackney/pdf/preview.rb]: NEEDS TO BE GONE.... but I'll take this out later

File: lib/hackney/pdf/preview_generator.rb [hunk 4]
```diff
@@ -1,36 +1,41 @@
...
 
       def execute(letter_params:)
-        params = validate_mandatory_fields(letter_params)
+        @letter = Hackney::ServiceCharge::Letter.new(letter_params)
+
+        @sender_address = ERB.new(File.open(SENDER_ADDRESS_PATH).read).result(binding)
+
+        @payment_options = ERB.new(File.open(PAYMENT_OPTIONS_PATH).read).result(binding)
+
+        @reply_form = ERB.new(File.open(REPLY_FORM_PATH).read).result(binding)
```
Comment (mrosel) [inline lib/hackney/pdf/preview_generator.rb]: I have never seen it done this way. However, I suppose it works and dry's it up.

**Pull request merged**